File: Examples/test-suite/matlab/argcargvtest_runme.m
largs={'hi','hola','hello'};
if (argcargvtest.mainc(largs) ~= 3)
  error('bad main typemap');
end

targs={'hi','hola'};
if (argcargvtest.mainv(targs,1) ~= 'hola')
  error('bad main typemap');
end

targs={'hi', 'hola'};
if (argcargvtest.mainv(targs,1) ~= 'hola')
  error('bad main typemap');
end

try
  error_flag = 0;
  argcargvtest.mainv('hello',1);
  error_flag = 1;
catch
end
if (error_flag)
  error('bad main typemap')
end


argcargvtest.initializeApp(largs);

File: Examples/test-suite/ruby/argcargvtest_runme.rb
#!/usr/bin/env ruby

require 'swig_assert'
require 'argcargvtest'

include Argcargvtest


$largs = ["hi", "hola", "hello"]
if mainc($largs) != 3
    raise RuntimeError, "bad main typemap"
end

$targs = ["hi", "hola"]
if mainv($targs, 0) != "hi"
    raise RuntimeError, "bad main typemap"
end
if mainv($targs, 1) != "hola"
    raise RuntimeError, "bad main typemap"
end
if mainv($targs, 2) != "<<NULL>>"
    raise RuntimeError, "bad main typemap"
end

$error = 0
$ret = 0
begin
    mainv("hello", 1)
    $error = 1
rescue => e
    $ret = 1
end

if $error == 1 or $ret != 1
    raise RuntimeError, "bad main typemap"
end

initializeApp($largs)

# Check that an empty array works.
$empty_args = []
if mainc($empty_args) != 0
    raise RuntimeError, "bad main typemap"
end
if mainv($empty_args, 0) != "<<NULL>>"
    raise RuntimeError, "bad main typemap"
end

# Check that empty strings are handled.
$empty_string = ["hello", "", "world"]
if mainc($empty_string) != 3
    raise RuntimeError, "bad main typemap"
end
if mainv($empty_string, 0) != "hello"
    raise RuntimeError, "bad main typemap"
end
if mainv($empty_string, 1) != ""
    raise RuntimeError, "bad main typemap"
end
if mainv($empty_string, 2) != "world"
    raise RuntimeError, "bad main typemap"
end
if mainv($empty_string, 3) != "<<NULL>>"
    raise RuntimeError, "bad main typemap"
end
